apps/bridge-dapp/src/containers/TxDetailContainer/TxDetailContainer.tsx

```diff
@@ -0,0 +1,167 @@
+import { type FC, useMemo, useState, useEffect } from 'react';
+import { Typography } from '@webb-tools/webb-ui-components';
+import { Note } from '@webb-tools/sdk-core';
+
+import InputOrOutputNotes from './InputOrOutputNotes';
+import SourceOrDestinationWalletInfo from './SourceOrDestinationWalletInfo';
+import TxBasicInfo from './TxBasicInfo';
+import { TxDetailContainerProps } from './types';
+
+const TxDetailContainer: FC<TxDetailContainerProps> = ({
+  hash,
+  activity,
+  amount,
+  noteAccountAddress,
+  walletAddress,
+  fungibleTokenSymbol,
+  wrappableTokenSymbol,
+  timestamp,
+  relayerName,
+  relayerFeeAmount,
+  inputNoteSerializations,
+  outputNoteSerializations,
+}) => {
+  const [inputNotes, setInputNotes] = useState<Note[]>([]);
+  const [outputNotes, setOutputNotes] = useState<Note[]>([]);
+
+  useEffect(() => {
+    async function getInputNotes() {
+      if (!inputNoteSerializations) return;
+      const inputNotes = await Promise.all(
+        inputNoteSerializations.map(async (serialization) => {
+          return await Note.deserialize(serialization);
+        })
+      );
+      setInputNotes(inputNotes);
+    }
+
+    getInputNotes();
+  }, [inputNoteSerializations]);
+
+  useEffect(() => {
+    async function getOutputNotes() {
+      if (!outputNoteSerializations) return;
+      const outputNotes = await Promise.all(
+        outputNoteSerializations.map(async (serialization) => {
+          return await Note.deserialize(serialization);
+        })
+      );
+      setOutputNotes(outputNotes);
+    }
+
+    getOutputNotes();
+  }, [outputNoteSerializations]);
+
+  const sourceTypedChainId = useMemo(() => {
+    // deposit
+    if (activity === 'deposit') {
+      return outputNotes.length > 0
+        ? +outputNotes[0].note.sourceChainId
+        : undefined;
+    }
+
+    // withdraw & transfer
```

Comment:
Are these comments accurate? They don't seem relevant to the code being commented. Also please use `// Capitals` when making comments.

Reply:
let me add more explanation there, @drewstone!